### PR TITLE
TedLium-V2 GMM system

### PR DESCRIPTION
**Pull request by christophmluscher**
still a few issues with the system:

- dependencies on users dir
- LM creation is buggy, so for now using a hard coded path

### CODE REVIEW

**Comment by christophmluscher** [comment]
this PR depends on #127 
should be merged after #127

**Review by Atticus1806 — DISMISSED** [review]
Blocking this for now because `BlissChangeEncodingJob` to wav crashes for me when running the pipeline because multi segment does not seem to be supported by `run_recover_duration`. I am not sure if adaption of the job is that simple.

Do we need this when converting to wav? I am not sure about the conversion process / lost precision here.

**Comment by Atticus1806** [comment]
Update: Pipeline runs, results are decent (13.4% WER on dev), only LM creation is missing und there is a bug in RASR (https://github.com/rwth-i6/rasr/pull/27). I will look into LM creation with Chris as a next step.

**Comment by JackTemaki** [comment]
> So as an update: It seems like disabling `run_recover_duration` does the job for now, but I am not sure about the implications and whether conversion to wav is safe. Maybe @JackTemaki knows?

No I do not know, but I think this is okay.

**Comment by Atticus1806** [comment]
Update: Pipeline runs, only LM creation is missing und there is a bug in RASR (https://github.com/rwth-i6/rasr/pull/27). I will look into LM creation with @christophmluscher  as a next step.

**Comment by michelwi** [comment]
> multi segment does not seem to be supported by run_recover_duration. I am not sure if adaption of the job is that simple.

The purpose of that function iirc is to set the correct ending time for all segments. Some corpora come with single segment recordings and have `start=0.0 end=inf` meaning "take the whole audio file". Then `run_recover_duration` puts the correct timestamp for end.

If there are multiple segments for one recording, then usually all have finite start/end times and the `run_recover_duration` task is unnecessary. (@Atticus1806 check if this is the case for you)

There could theoretically be the case where the first n-1 segments have finite times and the last one has `end=inf`. Then this currently breaks but the fix should be straightforward.

**Comment by Atticus1806** [comment]
`run_recover_duration task is unnecessary`
Grepping for `inf` on the corpus only yields segments so this should be fine.

**Comment by christophmluscher** [comment]
The pipeline depends on the jobs from this PR: rwth-i6/i6_core#406

**Review by JackTemaki — CHANGES_REQUESTED** [review]

File: common/setups/rasr/rasr_system.py
Comment (christophmluscher) [reply common/setups/rasr/rasr_system.py:136]: there are several options when to set the LM scale. normally this should be set in the recognition params and not here.

File: common/baselines/tedlium2/gmm/baseline_args.py
Comment (christophmluscher) [reply common/baselines/tedlium2/gmm/baseline_args.py:43]: we can still change this.. still needs to run for `common/asr` ^^

File: common/baselines/tedlium2/gmm/baseline_args.py
Comment (christophmluscher) [reply common/baselines/tedlium2/gmm/baseline_args.py:96]: for tedlium no.... 
@Atticus1806 did you??

Comment (Atticus1806) [reply common/baselines/tedlium2/gmm/baseline_args.py:96]: no, me neither.

Comment (christophmluscher) [reply common/baselines/tedlium2/gmm/baseline_args.py]: yes we reduced

File: common/setups/lm/srilm_system.py
Comment (JackTemaki) [inline common/setups/lm/srilm_system.py:343]: Run might not be a nice name to not confuse this with a job run.

File: common/setups/lm/srilm_system.py
Comment (JackTemaki) [inline common/setups/lm/srilm_system.py:162]: What does "katz" mean? In general, documentation of functions would be nice.

File: common/setups/lm/util.py
Comment (JackTemaki) [inline common/setups/lm/util.py:38]: Hmm, we still have the problem of where dataset helpers are located. Maybe it would be good to collect them all under `i6_experiments/common/setups/returnn` like I wanted to have in the beginning (instead of returnn-common). But moving this is not hash critical, so we could do that later.

**Review by JackTemaki — CHANGES_REQUESTED** [review]

File: common/setups/lm/util.py
Comment (Atticus1806) [reply common/setups/lm/util.py:38]: I would postpone this for now.

Comment (christophmluscher) [reply common/setups/lm/util.py:38]: > I would postpone this for now.

Agreed. I think we could put this on the agenda for our Tuesday meeting

**Review by JackTemaki — APPROVED** [review]
As this is urgent now, I allow merging this before the related SRILM jobs are merged.

**Pull request merged**
